docs/RationaleFIRRTL.md

```diff
@@ -42,6 +42,41 @@ Some of these may be research efforts that didn't gain broad adoption, in which
 case we don't want to support them.  However, if there is a good reason and a
 community that would benefit from adding support for these, we can do so.
 
+## Naming
+
+One of the goals of the FIRRTL compiler is to produce human-readable Verilog
+which can be easily mapped back to the input FIRRTL. Part of this effort
+means that we want the naming of Verilog objects to match the names used in
+the original FIRRTL, and they are predictably transformed during lowering.
+For example, after bundles are replaced with scalars in the lower-types pass,
+each field should be prefixed with the bundle name:
+
+```firrtl
+circuit Example
+  module Example
+    reg myreg: { a :UInt<1>, b: UInt<1> }, clock
+; firrtl-lower-types =>
+circuit Example
+  module Example
+    reg myreg_a: UInt<1>, clock
+    reg myreg_b: UInt<1>, clock
+```
+
+The name transformations applied by the SFC have become part of the
+documented API, and people rely on the final names to take a certain form.
+
+There are names for temporaries generated by the Chisel and FIRRTL tooling
+which are not important to maintain. These names are discarded when parsing,
+and new names are generated at Verilog export time. This has the effect of
+renumbering intermediate values and saves on memory during the compilation
+process. Names generated by Chisel typically look like `_T_12`, and names
+generated by the SCF look like `_GEN_12`. The FIRRTL compiler will not
+discard these names if the object has been annotated with `DontTouch`.
```

Comment:
It would be good to explain `DontTouch` with either: (1) explain what annotation means here or (2) just say an attribute with `class = firrtl.transforms.DontTouchAnnotation`. I'd lean towards (2) as (1) requires a section on annotations and how we're handling them.

Annotation is a name collision with more normal software language things and this has a very narrow meaning for FIRRTL.